HW04.Rmd

```diff
@@ -1,126 +1,239 @@
+# gun_deaths <- gun_deaths %>% mutate(race = replace(race, race == "Not Availabletive American/Not Availabletive Alaskan", "Native AmericaNative"))
+# It just replaces my string with NA, not the new string. :(
+
+# So I did this instead:
+gun_deaths <- gun_deaths %>% mutate(race = gsub("Not Availabletive American/Not Availabletive Alaskan", "Native American/Native Alaskan", race))
+gun_deaths <- gun_deaths %>% mutate(race = gsub("/", "/\n", race))
+
+```
+
+## Generating a data frame that summarizes the number of gun deaths per month and printing using `kable()`. 
+
+kable is part of the knitr package. Allows you to change column names and add captions and make pretty tables in your knitted document. Hint: set the argument format = "markdown"
+
+
+```{r month, warning=FALSE}
+deaths_per_month <- gun_deaths %>% count(month)
+
+knitr::kable(deaths_per_month, format = "markdown")
+```
+
+
+### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
+
+```{r month_chart}
+ggplot(deaths_per_month, aes(month.abb[month], n)) +
+  geom_bar(stat = "identity") +
+  xlab("Month") +
+  ylab("Gun Deaths") +
+  scale_x_discrete(limits = month.abb)
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+deaths_per_intent <- gun_deaths %>% count(intent)
+
+ggplot(deaths_per_intent, aes(reorder(intent, -n), n)) +
+  geom_bar(stat = "identity") +
+  xlab("Intent") +
+  ylab("Gun Deaths")
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+ggplot(gun_deaths, aes(sex, age)) +
+  geom_boxplot()
+```
+
+```{r female avg age}
+avg_age_f <- gun_deaths %>% 
+  group_by(sex) %>% 
+  summarize(average_age = mean(age, na.rm=TRUE)) %>%
+  filter(sex == "F")
+
+paste("Average age of female gun death victims:", avg_age_f[[2]])
+```
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+```{r education}
+count_m_white_hs <- gun_deaths %>%
+  filter(race == "White" & 
+           sex == "M" & 
+           education != "Less than HS" & 
+           year == 2012) %>%
+  count()
+
+paste("Number of white males with at least a HS education killed by guns in 2012:", 
+      count_m_white_hs[[1]])
+```
+
+
+## Which season of the year has the most gun deaths?
+
+Assume that:
+
+* Winter = January-March
+* Spring = April-June
+* Summer = July-September
+* Fall = October-December
+
+```{r season}
+gun_deaths_top_season <- fct_collapse(month.abb[gun_deaths$month],
+  Winter = c("Jan", "Feb", "Mar"),
+  Spring = c("Apr", "May", "Jun"),
+  Summer = c("Jul", "Aug", "Sep"),
+  Fall = c("Oct", "Nov", "Dec")) %>%
+  fct_count() %>%
+  top_n(1, n)
+
```

Comment:
nice solution to this question. I did not think about top_n, and the paste function is handy: makes the answers look polished.